### ISSUE TITLE
Examples of how to use it? 

### ISSUE THREAD

**Issue by sourcefrog** [openssh-rust/openssh-sftp-client#5]
Hi, nice to see this library.

It looks like it is meant to be used over the top of an SSH transport implementation that does authentication and encryption? Could you give some example code of how to connect it to OpenSSH or whatever transport is supported?

**Comment by NobodyXu** [comment]
> Hi, nice to see this library.
> 
> It looks like it is meant to be used over the top of an SSH transport implementation that does authentication and encryption? Could you give some example code of how to connect it to OpenSSH or whatever transport is supported?

Yes.

It can be used with [`openssh`](https://github.com/openssh-rust/openssh) once https://github.com/openssh-rust/openssh/pull/66 is merged in, where I wil refactor this crate so that you can use this crate by providing one input and one output stream.

**Comment by NobodyXu** [comment]
@sourcefrog [`openssh-sftp-client-0.10.0`](https://docs.rs/openssh-sftp-client/0.10.0/openssh_sftp_client/) is released with the refactor done.

Now you can simply pass stdin/stdout of the remote `sftp` subsystem to this [highlevel::Sftp::new](https://docs.rs/openssh-sftp-client/0.10.0/openssh_sftp_client/highlevel/struct.Sftp.html#method.new).

It will not be long before https://github.com/openssh-rust/openssh/pull/66 is also merged in so that you can use this crate with [openssh](https://github.com/openssh-rust/openssh).

**Comment by NobodyXu** [comment]
> Background task failed: read/flush task failed.

This means that the stdin/stdout is closed.

@elwerene Did you accidentally drop `child` before `Sftp::close` is called?

**Comment by elwerene** [comment]
You are right, I made a connect method and did not return the child so it went out of scope 🤦
Thanks!

**Comment by NobodyXu** [comment]
@sourcefrog [`openssh-0.9.1`] has released, with support for sftp subsystem, which can be used with [`openssh-sftp-client`].

[`openssh-0.9.1`]: https://docs.rs/openssh/0.9.1/openssh/
[`openssh-sftp-client`]: https://docs.rs/openssh-sftp-client/latest/openssh_sftp_client/
